### PR TITLE
Initial checkin of the monorepo structure + sw-appcache-behavior

### PR DESCRIPTION
**Pull request by jeffposnick**
R: @addyosmani @gauntface @wibblymat 
CC: @jakearchibald @jpmedley

Here's a (very large—sorry!) code dump of where I currently stand with the monorepo infrastructure + the first project, `sw-appcache-behavior`. There are some things that are still a work-in-progress, but it's far enough along to start breaking up additions into follow-up PRs.

Monorepo Implementation Status
- Build tasks.
- Test tasks.
- Linting tasks.
- **NOT FINISHED** Automatic documentation generation.
- **NOT FINISHED** Automatic `npm` publishing.

sw-appcache-behavior Status
- Functionality appears to be good.
- Initial set of end-to-end tests of caching behavior all pass.
- Basic samples.
- **NOT FINISHED** Additional end-to-end tests.
- **NOT FINISHED** Documentation.

The `sw-appcache-behavior` code was reviewed in https://github.com/jeffposnick/appcache2serviceworker/pull/1

### CODE REVIEW

File: build-utils.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import childProcess from 'child_process';
+import path from 'path';
+import promisify from 'promisify-node';
+
+export let globPromise = promisify('glob');
```
Comment (wibblymat) [inline build-utils.js]: Given that this is a style thing, and we have time constraints, I wouldn't block this but I'm opening a separate issue to discuss naming conventions. #2

**Comment by wibblymat** [comment]
I have ignored the stuff in `projects/sw-appcache-behavior/` because it was reviewed elsewhere.

If that's a correct assumption then LGTM.

File: npm-shrinkwrap.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (jeffposnick) [inline npm-shrinkwrap.json]: See https://github.com/GoogleChrome/sw-helpers/issues/5

File: build-utils.js (hunk 1 above)
Comment (jeffposnick) [inline build-utils.js]: ACK.

**Comment by jeffposnick** [comment]
Merging, and will put together some follow-up PRs to resolve repo issues not addressed in this PR.

**Comment by mattiaswelander** [comment]
Hi, I am not familiar with how service workers work, I have an application that uses appcache. Is this the right tool to convert my current application to use service workers instead? Is there any how-to site or document or am I just expected to drill through your code and also to learn everything about service workers before I can get started? Very new to this side of things.

**Comment by gauntface** [comment]
@mattiaswelander please raise questions here: https://github.com/GoogleChrome/sw-appcache-behavior

**Pull request merged**